src/OpenTelemetry.Exporter.InMemory/InMemoryExporterLoggingExtensions.cs

```diff
@@ -27,7 +27,11 @@ public static OpenTelemetryLoggerOptions AddInMemoryExporter(this OpenTelemetryL
             Guard.ThrowIfNull(loggerOptions);
             Guard.ThrowIfNull(exportedItems);
 
-            return loggerOptions.AddProcessor(new SimpleLogRecordExportProcessor(new InMemoryExporter<LogRecord>(exportedItems)));
+            return loggerOptions.AddProcessor(
+                new SimpleLogRecordExportProcessor(new InMemoryExporter<LogRecord>(exportedItems))
+                {
+                    BufferLogData = true,
```

Comment:
Is this for InMemoryExporter scenario, or there are more use cases?

Reply:
Check out the comment above this one. I was typing that out while you were reviewing 😄 I think it has general use, which is why I made it public. But it doesn't have to be, we could make it internal and use `InternalsVisibleTo` to allow `InMemoryExporter` project to set it. Or figure out something else entirely!

Reply:
I'm interested about the general use.

I thought InMemoryExporter is the only odd ball here because it allows users to examine the log records after `Export()` has returned. Do you see other cases where log records are consumed outside `Export()`?

Reply:
@reyang I couldn't up with any good concrete use cases for it to be public so I redid the change. Take a look and let me know what you think. I gave the log in-memory registration extension its own processor to control the buffering. No impact to actual workloads that way. Used InternalsVisibleTo in order to allow that processor to call `Buffer` without a public API change. I did try using the batch processor, but that changed the semantics. Tests need to spin or call force flush I think keeping it on simple keeps the tests simple 😉